README.md

```diff
@@ -89,8 +120,111 @@ browse order history, etc.
+
+### Expose via a Load Balancer                               
+
+1. Create the Load Balancer
+
+    ```bash
+    $ kubectl apply -f k8s/optional/ingress-controller.yaml 
+    
+    $ kubectl get services -n ingress-nginx
+    NAME            TYPE           CLUSTER-IP       EXTERNAL-IP       PORT(S)                      AGE
+    ingress-nginx   LoadBalancer   AAA.BBB.CCC.DDD   WWW.XXX.YYY.ZZZ  80:31475/TCP,443:30578/TCP   17s    
+    ```      
+
+    Once you have been assigned an external IP address, continue to the next step.              
+
+1. Setup Domains
+
+    You must have access to a top level domain for which you can create sub-domains to 
+    allow access to the application via a Load Balancer (LB).
+
+    For example if your top level domain is `mycompany.com` then you
+    should create a single wildcard DNS entry `*.sockshop.mycompany.com` to 
+    point to your external LB IP address.
+    
+1. Create the ingress
+
+    Each time you use a different back-end you will need to create a new ingress.
+
+    In your terminal, export (or SET for Windows) your top level domain
+    and the backend you are using. 
+    
+    For example for domain `sockshop.mycompany.com` and `memory` backend, use the following 
+    
+    ```bash
+    $ export SOCKSHOP_DOMAIN=sockshop.mycompany.com
+    $ export SOCKSHOP_BACKEND=memory                            
+    
+    $ envsubst -i k8s/optional/ingress.yaml| kubectl apply -f -
+    
+    $ kubectl get ingress  
+    
+    NAME               HOSTS                                                                                         ADDRESS           PORTS   AGE
+    mp-ingress         mp.memory.mycompany.com                                                                       XXX.XXX.XXX.XXX   80      12d
+    sockshop-ingress   memory.sockshop.mycompany.com,jaeger.memory.sockshop.mycompany.com,api.memory.mycompany.com   XXX.XXX.XXX.XXX   80      12d
+    ```         
+
+1. Access the application
+
+    Access the application via the endpoint http://memory.sockshop.mycompany.com/ 
+
+1. Cleanup the ingress
+
+    To cleanup the ingress for your deployment, issue the following ensuring you have the same
+    environment variables set from when you created the ingress. 
+
+    ```bash 
+    $ export SOCKSHOP_DOMAIN=sockshop.mycompany.com  
+    $ export SOCKSHOP_BACKEND=memory                             
+    
+    $ envsubst -i k8s/optional/ingress.yaml| kubectl delete -f -
+    ``` 
+    
+1. Remove the LB (Optional)
+    
+    If you wish to remove your LB, issue the following
+    
+    ```bash
+    $ kubectl delete -f k8s/optional/ingress-controller.yaml
+    ```
+
+### Configure Jaegar
+
+```bash
+$ kubectl create -f k8s/optional/jaeger-operator.yaml 
+
+$ kubectl create -f k8s/optional/jaegar.yaml
+```                                         
+
+Access the application via the endpoint http://jaegar.memory.sockshop.mycompany.com/ 
+
+
+### Cleanup the Extensions
```

Comment:
"Cleanup"

I would provide the steps to clean everything up here, including the services.